server/protocols/jmap/src/main/java/org/apache/james/jmap/UserProvisioningFilter.java

```diff
@@ -75,11 +73,11 @@ void createAccountIfNeeded(MailboxSession session) {
         try {
             User user = session.getUser();
             if (needsAccountCreation(user)) {
-                createAccount(user);
+                usersRepository.addUser(getUsername(user), generatePassword());
```

Comment:
I don' t really agree, I concidered it as a useless method call. Here we clearly see we add a user...

Moreover **Account** is a piece of domain knowledge that is used nowhere else in James, I find it confusing...